src/hotspot/share/opto/macro.cpp

```diff
@@ -1393,6 +1394,9 @@ void PhaseMacroExpand::expand_allocate_common(
   // Copy debug information and adjust JVMState information, then replace
   // allocate node with the call
   call->copy_call_debug_info(&_igvn, alloc);
+  if (valid_length_test != NULL) {
```

Comment:
How about adding a comment or assert that makes it clear that valid_length_test != NULL if it's an array that's allocated? If length == NULL, then valid_length_test is NULL too. Just to ease the reading of this code a bit.

Reply:
Thanks for looking at this.
What about the comment I just pushed?